freedomofpress/securedrop-client
# [0.14.0] Truncate source list displayed in delete dialog

## Status

Ready for review

## Description

Fixes #2284

Sets a limit of 32 sources to be displayed in the deletion dialog before the list is truncated to avoid text overruns that push the safety warnings out of the viewport.
- If the number of sources is 32 or less, the entire list of sources is displayed
- If the number is 33 or more, 30 sources are displayed followed by a message `... plus N additional sources` where N = len(sources -30)  

## Test Plan

- set up an SD instance (dev easiest) with ~150 sources.
- start the client using that instance
- [ ] select 1 source, click DELETE SOURCES and verify that only the source is shown. Click "YES.." and verify that the source is deleted.
 
- [ ] select 32 sources, click DELETE SOURCES and verify that the sources are all shown. Click "YES.." and verify that the sources are deleted.

- [ ] select 33 sources, click DELETE SOURCES and verify that the first 30 sources are shown followed by a message `plus 3 additional sources`. Click "YES.." and verify that the sources are deleted.
-[ ] select all remaining sources, click DELETE SOURCES and verify that the first 30 sources are shown followed by a message `plus N additional sources`, where N+30 equals the total number of sources shown in the `YES,...` button . Click "YES.." and verify that all sources are deleted.
